packages/core/lib/debug/printer.js

```diff
@@ -593,14 +606,15 @@ class DebugPrinter {
 
     this.config.logger.log();
 
-    for (const [section, variables] of Object.entries(sections)) {
-      if (variables.length > 0) {
+    for (const [section, sections] of Object.entries(sectionKeys)) {
+      const printThisSection = sectionOuts.has(section.slice(0,3));
```

Comment:
add a comment:
// printout the sections that are included in the inputs and have positive contents length

Reply:
Ah, sorry, that's not what I meant.  I meant, it might be worth explaining the expression `sectionOuts.has(section.slice(0,3))`, because a person reading the code might have to think about that one.  I was implicitly referring to our earlier discussion where I mentioned that; sorry, I possibly should have made that explicit.

Reply:
Again, that's if you think it's worth explaining; if you don't think it needs it, you don't have to bother.

Reply:
add a comment
```
      // only check the first 3 characters of each name given in the input sectionPrintouts
      // since each section name defined in the constructor contains 3 characters
      const printThisSection = sectionOuts.has(section.slice(0,3));
```

Reply:
Thanks, I think this helps!